Content.Server/Cloning/CloningConsoleSystem.cs

```diff
@@ -0,0 +1,312 @@
+                        {
+                            clonerStatus = ClonerStatus.NoMindDetected;
+                        }
+                    }
+                }
+            }
+
+            // cloning pod info
+            var cloneBodyInfo = "Unknown";
+            float cloningProgress = 0;
+            float cloningTime = 30f;
+            bool clonerProgressing = false;
+            bool clonerConnected = false;
+            bool clonerMindPresent = false;
+            if (consoleComponent.CloningPod != null && TryComp<CloningPodComponent>(consoleComponent.CloningPod, out var clonePod))
+            {
+                clonerConnected = true;
+                EntityUid? cloneBody = clonePod.BodyContainer.ContainedEntity;
+                if (TryComp<MetaDataComponent>(cloneBody, out var cloneMetaData))
+                    cloneBodyInfo = cloneMetaData.EntityName;
+
+                cloningProgress = clonePod.CloningProgress;
+                cloningTime = clonePod.CloningTime;
+                clonerProgressing = _cloningSystem.IsPowered(clonePod) && (clonePod.BodyContainer.ContainedEntity != null);
+                clonerMindPresent = clonePod.Status == CloningPodStatus.Cloning;
+                if (cloneBody != null)
+                {
+                    clonerStatus = ClonerStatus.ClonerOccupied;
+                }
+            }
+            else
+            {
+                clonerStatus = ClonerStatus.NoClonerDetected;
+            }
+
+            return new CloningConsoleBoundUserInterfaceState(
+                scanBodyInfo,
+                cloneBodyInfo,
+                _gameTiming.CurTime,
+                cloningProgress,
+                cloningTime,
+                clonerProgressing,
+                clonerMindPresent,
+                clonerStatus,
+                scannerConnected,
+                clonerConnected
+                );
+        }
+
+        public void DisconnectMachineConnections(EntityUid uid, CloningConsoleComponent? consoleComponent)
+        {
+            if (!Resolve(uid, ref consoleComponent))
+                return;
+
+            if (consoleComponent.CloningPod != null && TryComp<CloningPodComponent>(consoleComponent.CloningPod, out var cloningPod) && cloningPod.ConnectedConsole == uid)
+                cloningPod.ConnectedConsole = null;
+
+            if (consoleComponent.GeneticScanner != null && TryComp<MedicalScannerComponent>(consoleComponent.GeneticScanner, out var medicalScanner) && medicalScanner.ConnectedConsole == uid)
+                medicalScanner.ConnectedConsole = null;
+
+            consoleComponent.CloningPod = null;
+            consoleComponent.GeneticScanner = null;
+        }
+
+        private HumanoidCharacterProfile GetPlayerProfileAsync(NetUserId userId)
+        {
+            return (HumanoidCharacterProfile)  _prefsManager.GetPreferences(userId).SelectedCharacter;
+        }
```

Comment:
I think should should be nullable and return null if the preferences for the session hasn't been loaded? Also is this even async?

Reply:
Yeah, this function is destroyed now with appearancecomponent changes thankfully.